alacritty/src/scheduler.rs

```diff
@@ -0,0 +1,114 @@
+//! Scheduler for emitting events at a specific time in the future.
+
+use std::collections::VecDeque;
+use std::time::{Duration, Instant};
+
+use glutin::event::Event as GlutinEvent;
+
+use crate::event::Event as AlacrittyEvent;
+
+/// Event ID for scrolling during selection.
+pub const SELECTION_SCROLLING_EVENT: u64 = 0;
```

Comment:
It's not event, it's a timer id. So you have some predefined timer ids and arbitrary timer. But if it's a predefined timer, how one could handle multiwindow? I'd assume no problem for selection, but some other things won't work. Anyway, the way it works right now kind of makes sense, however now I wonder what's the idea of `next_id` on `Scheduler` if we can just have a set of predefined timers, and so you never touch `next_id` anyway.

Also, we all is about predefined times, then it should be enum, I guess.

Reply:
Good point, since we do not need next_id yet, I'll convert it to an enum. If we decide that we need it later, we can always just change it.